src/main/java/de/tum/in/www1/artemis/service/LectureUnitProcessingService.java

```diff
@@ -0,0 +1,205 @@
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.multipart.commons.CommonsMultipartFile;
+
+import de.tum.in.www1.artemis.domain.Attachment;
+import de.tum.in.www1.artemis.domain.enumeration.AttachmentType;
+import de.tum.in.www1.artemis.domain.lecture.AttachmentUnit;
+import de.tum.in.www1.artemis.web.rest.dto.LectureUnitDTO;
+import de.tum.in.www1.artemis.web.rest.dto.LectureUnitInformationDTO;
+import de.tum.in.www1.artemis.web.rest.dto.LectureUnitSplitDTO;
+import de.tum.in.www1.artemis.web.rest.errors.InternalServerErrorException;
+
+@Service
+public class LectureUnitProcessingService {
+
+    private final Logger log = LoggerFactory.getLogger(LectureUnitProcessingService.class);
+
+    /**
+     * Split units from given file according to given split information.
+     * @param lectureUnitSplitDTOs The split information
+     * @param file The file (lecture slide) to be split
+     * @return The prepared units to be saved
+     */
+    public List<LectureUnitDTO> splitUnits(List<LectureUnitSplitDTO> lectureUnitSplitDTOs, MultipartFile file) throws IOException {
+
+        try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream(); PDDocument document = PDDocument.load(file.getBytes())) {
+            List<LectureUnitDTO> units = new ArrayList<>();
+            Splitter pdfSplitter = new Splitter();
+
+            for (LectureUnitSplitDTO lectureUnit : lectureUnitSplitDTOs) {
+                AttachmentUnit attachmentUnit = new AttachmentUnit();
+                Attachment attachment = new Attachment();
+                PDDocumentInformation pdDocumentInformation = new PDDocumentInformation();
+
+                pdfSplitter.setStartPage(lectureUnit.startPage());
+                pdfSplitter.setEndPage(lectureUnit.endPage());
+                pdfSplitter.setSplitAtPage(lectureUnit.endPage());
+
+                List<PDDocument> documentUnits = pdfSplitter.split(document);
+                pdDocumentInformation.setTitle(lectureUnit.unitName());
+                documentUnits.get(0).setDocumentInformation(pdDocumentInformation);
+                documentUnits.get(0).save(outputStream);
+
+                // setup attachmentUnit and attachment
+                attachmentUnit.setDescription("");
+                attachment.setName(lectureUnit.unitName());
+                attachment.setAttachmentType(AttachmentType.FILE);
+                attachment.setReleaseDate(lectureUnit.releaseDate());
+                attachment.setUploadDate(ZonedDateTime.now());
+
+                MultipartFile multipartFile = convertByteArrayToMultipart(lectureUnit.unitName(), outputStream.toByteArray());
+
+                LectureUnitDTO lectureUnitsDTO = new LectureUnitDTO(attachmentUnit, attachment, multipartFile);
+                units.add(lectureUnitsDTO);
+                documentUnits.get(0).close();
+            }
+
+            outputStream.close();
+            document.close();
```

Comment:
You don't need to manually close streams when using try-with-resources, they will get automatically closed.
```suggestion
```